### PR TITLE
Compatibility for 0.7/1.0

### PR DESCRIPTION
**Pull request by samuelpowell**
Uses Compat to retain backwards compatibility with 0.6.

### CODE REVIEW

**Review by ReubenHill — CHANGES_REQUESTED** [review]
Compiles and runs fine on 0.6.x. On 0.7.x get a number of deprecation warnings when running the example in README.md:
```Julia
julia> using LibFTD2XX

julia> devs = createdeviceinfolist()
0x00000004

julia> list, elnum = getdeviceinfolist(devs)
┌ Warning: `Vector{T}(m::Integer) where T` is deprecated, use `Vector{T}(undef, m)` instead.
│   caller = getdeviceinfolist(::UInt32) at LibFTD2XX.jl:109
└ @ LibFTD2XX C:\Users\Reuben\.julia\packages\LibFTD2XX\ImzKb\src\LibFTD2XX.jl:109
(LibFTD2XX.FT_DEVICE_LIST_INFO_NODE[FT_DEVICE_LIST_INFO_NODE(0x00000002, 0x00000007, 0x04036011, 0x00000000, (70, 84, 50, 75, 72, 49, 72, 49, 65, 0, -107, 15, -6, 127, 0, 0), (85, 83, 66, 32, 60, 45, 62, 32, 83, 101, 114, 105, 97, 108, 32, 67, 111, 110, 118, 101, 114, 116, 101, 114, 32, 65, 0, 0, 0, 0, 0, 0, -52, 120, 12, -71, 26, -100, 0, 0, 64, 36, 16, 43, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -106, 86, -17, 12, -6, 127, 0, 0), Ptr{Nothing} @0x0000000000000000), FT_DEVICE_LIST_INFO_NODE(0x00000002, 0x00000007, 0x04036011, 0x00000000, (70, 84, 50, 75, 72, 49, 72, 49, 66, 0, -48, 44, 0, 0, 0, 0), (85, 83, 66, 32, 60, 45, 62, 32, 83, 101, 114, 105, 97, 108, 32, 67, 111, 110, 118, 101, 114, 116, 101, 114, 32, 66, 0, -128, 1, 0, 0, 0, -16, -57, 107, 41, 0, 0, 0, 0, -96, 12, -48, 44, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, -16, -57, 107, 41, 0, 0, 0, 0), Ptr{Nothing} @0x0000000000000000), FT_DEVICE_LIST_INFO_NODE(0x00000002, 0x00000007, 0x04036011, 0x00000000, (70, 84, 50, 75, 72, 49, 72, 49, 67, 0, -48, 44, 0, 0, 0, 0), (85, 83, 66, 32, 60, 45, 62, 32, 83, 101, 114, 105, 97, 108, 32, 67, 111, 110, 118, 101, 114, 116, 101, 114, 32, 67, 0, -128, 1, 0, 0, 0, 48, -51, 107, 41, 0, 0, 0, 0, 8, 13, -48, 44, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 48, -51, 107, 41, 0, 0, 0, 0), Ptr{Nothing} @0x0000000000000000), FT_DEVICE_LIST_INFO_NODE(0x00000002, 0x00000007, 0x04036011, 0x00000000, (70, 84, 50, 75, 72, 49, 72, 49, 68, 0, -48, 44, 0, 0, 0, 0), (85, 83, 66, 32, 60, 45, 62, 32, 83, 101, 114, 105, 97, 108, 32, 67, 111, 110, 118, 101, 114, 116, 101, 114, 32, 68, 0, -128, 1, 0, 0, 0, -16, -60, 107, 41, 0, 0, 0, 0, 112, 13, -48, 44, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, -16, -60, 107, 41, 0, 0, 0, 0), Ptr{Nothing} @0x0000000000000000)], 0x00000004)

julia> description = String(list[3].description)
┌ Warning: `find` is deprecated, use `findall` instead.
│   caller = String(::NTuple{64,Int8}) at LibFTD2XX.jl:90
└ @ LibFTD2XX C:\Users\Reuben\.julia\packages\LibFTD2XX\ImzKb\src\LibFTD2XX.jl:90
"USB <-> Serial Converter C"

julia> handle = ftopen(0)
┌ Warning: `finalizer(o, f::Function)` is deprecated, use `finalizer(f, o)` instead.
│   caller = Type at LibFTD2XX.jl:61 [inlined]
└ @ Core C:\Users\Reuben\.julia\packages\LibFTD2XX\ImzKb\src\LibFTD2XX.jl:61
FT_HANDLE(Ptr{Nothing} @0x00000000296bc7f0)

julia> isopen(handle)
true

julia> close(handle)

julia> handle = open(description, OPEN_BY_DESCRIPTION)
┌ Warning: `finalizer(o, f::Function)` is deprecated, use `finalizer(f, o)` instead.
│   caller = Type at LibFTD2XX.jl:61 [inlined]
└ @ Core C:\Users\Reuben\.julia\packages\LibFTD2XX\ImzKb\src\LibFTD2XX.jl:61
FT_HANDLE(Ptr{Nothing} @0x00000000296bc1f0)

julia> write(handle, Vector{UInt8}("Hello"))
┌ Warning: Vector{UInt8}(s::String) will copy data in the future. To avoid copying, use `unsafe_wrap` or `codeunits` instead.
│   caller = top-level scope at none:0
└ @ Core none:0
0x00000005

julia> nb_available(handle)
0x00000005

julia> String(read(handle, 5))
"\x80\x0f\e\x03\0"

julia> write(handle, Vector{UInt8}("world!"))
┌ Warning: Vector{UInt8}(s::String) will copy data in the future. To avoid copying, use `unsafe_wrap` or `codeunits` instead.
│   caller = top-level scope at none:0
└ @ Core none:0
0x00000006

julia> String(readavailable(handle))
┌ Warning: `Vector{T}(m::Integer) where T` is deprecated, use `Vector{T}(undef, m)` instead.
│   caller = readavailable(::FT_HANDLE) at LibFTD2XX.jl:205
└ @ LibFTD2XX C:\Users\Reuben\.julia\packages\LibFTD2XX\ImzKb\src\LibFTD2XX.jl:205
"\x01\0\0\0\0\0"

julia> flush(handle)

julia> close(handle)
"USB <-> Serial Converter C"

julia> handle = ftopen(0)
┌ Warning: `finalizer(o, f::Function)` is deprecated, use `finalizer(f, o)` instead.
│   caller = Type at LibFTD2XX.jl:61 [inlined]
└ @ Core C:\Users\Reuben\.julia\packages\LibFTD2XX\ImzKb\src\LibFTD2XX.jl:61
FT_HANDLE(Ptr{Nothing} @0x00000000296bc7f0)

julia> handle = open(description, OPEN_BY_DESCRIPTION)
┌ Warning: `finalizer(o, f::Function)` is deprecated, use `finalizer(f, o)` instead.
│   caller = Type at LibFTD2XX.jl:61 [inlined]
└ @ Core C:\Users\Reuben\.julia\packages\LibFTD2XX\ImzKb\src\LibFTD2XX.jl:61
FT_HANDLE(Ptr{Nothing} @0x00000000296bc1f0)

julia> write(handle, Vector{UInt8}("Hello"))
┌ Warning: Vector{UInt8}(s::String) will copy data in the future. To avoid copying, use `unsafe_wrap` or `codeunits` instead.
│   caller = top-level scope at none:0
└ @ Core none:0
0x00000005

**Comment by samuelpowell** [comment]
I don't think you're on the right branch on 0.7 (perhaps not even on 0.6) because I fixed those

**Comment by ReubenHill** [comment]
Strange, I did `(v0.7) pkg> add https://github.com/Gowerlabs/LibFTD2XX.jl#sp/07`

**Comment by samuelpowell** [comment]
Well I don't know how broaches work when you add them, which is really for using them. I would suggest you develop...
`(v0.7) pkg> dev https://github.com/Gowerlabs/LibFTD2XX.jl#sp/07`

This actually gets the git and puts it in `.julia/dev` for development work

**Comment by ReubenHill** [comment]
NB, that doesn't work
```Julia
(v0.7) pkg> dev https://github.com/Gowerlabs/LibFTD2XX.jl#sp/07
ERROR: a git revision cannot be given to `develop`
```

Anyway, I manually switched branches and all looks good. The example code throws conversion warnings
```Julia
julia> write(handle, Vector{UInt8}("Hello"))
```
warns 
```Julia
Warning: Vector{UInt8}(s::String) will copy data in the future. To avoid copying, use `unsafe_wrap` or `codeunits` instead.
```
so the README.md might need updating but otherwise all looks good.

**Comment by samuelpowell** [comment]
Yeah that's not a package problem per se but feel free to push a change to the README

**Comment by samuelpowell** [comment]
I do it

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Julia wrapper for FTD2XX driver. For reference see the [D2XX Programmer's Guide]
 The below is a demonstration for a port which echos what it received.
 
 ```Julia
-julia> using LibFTD2XX
+julia> using Compat, LibFTD2XX
```
Comment (ReubenHill) [inline README.md]: Why `using Compat`?

Comment (samuelpowell) [reply README.md]: For code units, which fails

File: README.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ julia> nb_available(handle)
 julia> String(read(handle, 5))
 "Hello"
 
-julia> write(handle, Vector{UInt8}("world!"))
+julia> write(handle, codeunits("world!"))
```
Comment (ReubenHill) [inline README.md]: throws error:
```Julia
julia> write(handle, codeunits("Hello"))
ERROR: FT_HANDLE does not support byte I/O
```
lack of byte IO support is outstanding issue

**Review by ReubenHill — APPROVED** [review]
Will update README in future and create issue RE byte IO for now. Thanks for your changes!

**Pull request merged**